### ISSUE TITLE
1password integration on OS X UI/UX confusing

### ISSUE THREAD

**Issue by andycutright** [brave/browser-laptop#1297]
After enabling 1password integration, via Brave preferences, I have no idea how to use 1password to fill the username/ password fields. 

How is this supposed to work? 

1password 3.x
OS X 10.10.5

**Comment by megalithic** [comment]
Right click an input field, for instance, and choose `1Password` from the context menu. However, you'll need to disable signature verification checks, @bbondy mentioned this setting lives in `Preferences` under an `Advanced` sub-setting, but I've not found it in the UI, nor in the source. :( Perhaps he can shed some light.

**Comment by andycutright** [comment]
Already tried right-clicking, and found the context menu, to no effect.

I don't see any Advanced setting, either.

**Comment by megalithic** [comment]
You should have at least been presented with the miniature 1password "master password" window (exactly like the one in chrome), but upon entering your master password, you should get a prompt complaining about the signature verification. 

Anyway, it's a non-issue until we're able to either a.) disable signature verification, or b.) 1password signs it :)

**Comment by megalithic** [comment]
ZOMG, I'm such an idiot; the settings @bbondy is referring to, lives within 1Password itself... I hang my head in shame.

**Comment by andycutright** [comment]
I was not presented with the master password window. Also, note I'm using
1password 3.x, as specified earlier. Not 4.x.

When you state there's a way to disable signature verification, do you mean
it can be disabled in 1password, or disabled in brave?

If the feature isn't working, maybe y'all should back it out. I assume you
work on brave.

**Comment by andycutright** [comment]
I see you meant 1password. That feature isn't present in 3.x, afaik

**Comment by megalithic** [comment]
@andycutright, nope, i'm just a user. have poked around the code some to see about adding vim-esque keyboard navigation (because why would you not want to navigate around the page with vim navigation keys).

**Comment by megalithic** [comment]
For the record, once disabling signature verification (I'm on 6.x), it works as intended. Though, another area to add keyboard shortcuts (1cmd + /` on osx, like in chrome).

Anyway, perhaps they just need to document that only 1password versions 4.x+ are supported. It's totally worth upgrading though. the Agilebits team are constantly adding features/fixing bugs, etc.

**Comment by megalithic** [comment]
oh and one last comment about the UI/UX; @bbondy has mentioned on his twitter feed (and in other release notes here on the repo) that they are working towards native chrome extension support, which sounds like this is the first step, so I'd imagine things will get better. No need to back out a feature that definitely works, but may be rough around the edges for 100% use cases. Anyway, good luck to you @andycutright!

**Comment by midi** [comment]
@andycutright, you can access it with 1PW mini for example<img width="1280" alt="screen shot 2016-04-09 at 12 01 31" src="https://cloud.githubusercontent.com/assets/1421440/14402901/ec5883e8-fe4a-11e5-8d7e-afb175b041b6.png">

**Comment by andycutright** [comment]
Are you a brave dev?

**Comment by midi** [comment]
@andycutright I'm a Brave user.

**Comment by andycutright** [comment]
What version of 1password are you running?

**Comment by midi** [comment]
6.2. Maybe contact 1Password and inquire if you'll also get the updated signatures with your version.

**Comment by diracdeltas** [comment]
We are waiting on 1password to ship our code signatures; right now, using 1pw with Brave requires disabling signature verification in 1pw, which is not recommended.

cc @bridiver - seems that `about:preferences` should be clearer about this.

**Comment by bridiver** [comment]
I thought about adding something, but it really requires instructions that are more detailed than what is practical to put there. We also don't know when exactly it will change and the information will not longer be relevant. @bbondy did put something in the release notes I believe and 1password opens a popup telling you that the browser is not signed when you try to use it

**Comment by bridiver** [comment]
I think the other issue is that we didn't really want to tell people to disable a security feature of 1pw. It doesn't open up a very big threat, but I'm still not sure that we want to recommend it. Unfortunately until 1pw releases with our signature I think it's an "expert-only" feature

**Comment by bridiver** [comment]
@midi - our understanding is that the mac code signature should be included with the next 1pw release. The Windows signature will apparently take longer for some reason.

**Comment by andycutright** [comment]
Ok, looking forward to the integration.

On Sun, Apr 10, 2016 at 1:44 PM, yan zhu (@bcrypt) <notifications@github.com

**Comment by diracdeltas** [comment]
Not everyone reads the release notes - maybe put a link next to the pref that links to a wiki page explaining the current steps needed to make it work? Especially if this will be a problem on windows for a while.

**Comment by webbj74** [comment]
Email from AgileBits support:

> We've actually just added support for Brave in the beta version of 1Password for Mac (as of 6.3.BETA-14)! If you are not already using the beta, here is how you can get it: https://support.1password.com/install-beta/mac.html. If you do choose to use the beta, we would love to hear from you if you have any feedback.

**Comment by halo** [comment]
>  If you do choose to use the beta, we would love to hear from you if you have any feedback.

Using 1Password `6.3.1` works just fine in Brave `0.10.3`.

**Comment by cezaraugusto** [comment]
after 1Password support pointed by @webbj74, is it still an issue?

**Comment by diracdeltas** [comment]
Closing for now, thanks

**Comment by bridiver** [comment]
one thing we are still missing is the browser action icon and I think that was really the original problem, but I'm going to open up a new ticket for that #2224
